### ISSUE TITLE
TEAS 0.7.0 is currently non-functioning

### ISSUE THREAD

**Issue by dandln** [dandln/TheEpicAudioStreamer#15]
There's a breaking change in the Discord Voice API that's being enforced on November 18 that will cause the voice channel component of DSharpPlus (the API implementation that TEAS is using) to stop working on that date. A new version that adapted to said API change was released today (implementing https://github.com/DSharpPlus/DSharpPlus/pull/2169), and I've migrated TEAS to that version, however while the bot itself works, it does not actually transmit any audio due to (what appears to be) a different issue with the library.

I've been in contact with devs of DSharpPlus and they can reproduce the issue but currently don't know what is causing it. In light of that, there will likely not be a new version of TEAS with a fix before the deadline, meaning that it will likely stop working once the API change is being enforced.

This is currently outside of my control, but I'll try to get a new version up as soon as there's a fix for the issue available from DSharpPlus and I've had time to implement it. If you've planned on using TEAS in the next couple of days, you might want to consider an alternative.

**Comment by ujjukumar** [comment]
Any update on this as the mentioned PR is now complete.

**Comment by dandln** [comment]
Yes and no, as mentioned above while the PR mentioned should have implemented the correct encryption methods, there seems to be a different issue which resulted in the library not actually transmitting any audio. This issue was reproducable by the library devs but they could not figure out what was causing it.

I can check in with them again after the holidays to see if there was any progress or another change in the library miraculously fixed this issue as a side-effect, but I'm not getting my hopes up. The voice functionality of the library is currently also undergoing a full rewrite anyway, so in worst case I will have to wait until that is complete.

**Comment by dandln** [comment]
No update so far, issue still persists and it remains a mystery to both the library devs and myself why. I will make sure to revisit when appropriately, but it's unlikely to be resolved any time soon.

**Comment by dandln** [comment]
I've had multiple reports right now that TEAS, in fact, did not stop working on the day of the deadline and does work still. I'm not sure why that is, but this means that you might be able to continue using it just fine. _If_ it breaks, though, I won't be able to troubleshoot/fix.

**Comment by dandln** [comment]
Quick update here (it's been a while :D ) - The DSharpPlus devs have mostly finished the rewrite of the Voice part of the library, and I'm currently testing the changes with TEAS. There's some issues still that I'm ironing out together with them now before I can say everything is working again, but we're definitely making progress, and that's good news!

**Comment by dandln** [comment]
This is _finally_ fixed in [v0.7.1](https://github.com/dandln/TheEpicAudioStreamer/releases/tag/v0.7.1) now! :)
